### PR TITLE
CMake: modernize, add metadata

### PR DESCRIPTION
**Pull request by scivision**
* add metadata to project()
* cmake_policy(set ... NEW) is implied for these policies by cmake_minimum_required(VERSION 3.15)
* string(APPEND ... helps avoid typos
* make tab indents everywhere in CMakeLists.txt as this seems to be the existing Ninja project practice

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,10 @@
 cmake_minimum_required(VERSION 3.15)
-cmake_policy(SET CMP0091 NEW)
-project(ninja)
+project(ninja
+LANGUAGES CXX
```
Comment (jonesmz) [inline CMakeLists.txt]: Indentation

**Comment by jonesmz** [comment]
I've not heard of the cmake string() function.

Is that newish? Or am I just out of the loop?

**Comment by jonesmz** [comment]
This change looks like a good improvement to me. We should merge it.

**Comment by scivision** [comment]
string(APPEND ...) was added in CMake 3.4

**Comment by jhasse** [comment]
What's the use-case for the metadata?

Regarding tab-indents: While it is used in CMakeLists.txt it might have been a mistake as the rest of the code base uses 2-spaces indent.

**Comment by scivision** [comment]
A primary use for the "description" and "homepage_url" are by [CPack](https://cmake.org/cmake/help/latest/module/CPack.html), which can be used to generate binary / source installers for various platforms.

**Comment by jhasse** [comment]
Not sure if I'd want to add CPack support.

Can you make a PR with only

> cmake_policy(set ... NEW) is implied for these policies by cmake_minimum_required(VERSION 3.15)

and

>string(APPEND ... helps avoid typos

?

**Comment by jhasse** [comment]
Thanks :)

**Pull request merged**